packages/@aws-cdk/aws-cloudfront/lib/public-key.ts

```diff
@@ -0,0 +1,81 @@
+import { Names, Resource, Token } from "@aws-cdk/core";
+import { Construct } from "constructs";
+import { CfnPublicKey } from "./cloudfront.generated";
+
+/**
+ * Represents a Public Key
+ */
+export interface IPublicKey {
+  /**
+   * The ID of the key group.
+   * @attribute
+   */
+  readonly publicKeyId: string;
+}
+
+/**
+ * Properties for creating a Public Key
+ */
+export interface PublicKeyProps {
+  /**
+   * A name to identify the key group.
+   * @default - generated from the `id`
+   */
+  readonly publicKeyName?: string;
+
+  /**
+   * A comment to describe the public key.
+   * @default - no comment
+   */
+  readonly comment?: string;
+
+  /**
+   * A string included in the request to help make sure that the request can’t be replayed.
+   */
+  readonly callerReference: string;
+
+  /**
+   * The public key that you can use with signed URLs and signed cookies, or with field-level encryption.
+   * @see https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/PrivateContent.html
+   * @see https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/field-level-encryption.html
+   */
+  readonly encodedKey: string;
+}
+
+/**
+ * A Public Key Configuration
+ * 
+ * @resource AWS::CloudFront::KeyGroup
+ */
+export class PublicKey extends Resource implements IPublicKey {
+  public static fromPublicKeyId(scope: Construct, id: string, publicKeyId: string): IPublicKey {
+    return new class extends Resource implements IPublicKey {
+      public readonly publicKeyId = publicKeyId;
+    }(scope, id);
+  }
+
+  public readonly publicKeyId: string;
+
+  constructor(scope: Construct, id: string, props: PublicKeyProps) {
+    super(scope, id, {
+      physicalName: props.publicKeyName,
+    });
+
+    const publicKeyName = props.publicKeyName ?? Names.uniqueId(this);
+    if (!Token.isUnresolved(publicKeyName) && !publicKeyName.match(/^[\w-]+$/i)) {
+      throw new Error(`'publicKeyName' can only include '-', '_', and alphanumeric characters, got: '${props.publicKeyName}'`);
+    }
+
+    const resource = new CfnPublicKey(this, 'Resource', {
+      publicKeyConfig: {
+        name: publicKeyName,
+        creationStack: [""],
```

Comment:
I don't see this prop in the generated code or the docs. Where did it come from?

Reply:
Hmm interesting... Yesterday my VS Code was complaining that this property is required... today it can be omitted. Weird. Must be because of my VS Code.

![image](https://user-images.githubusercontent.com/31543/106162618-8b249400-6145-11eb-8d7c-9a668ed3955a.png)

Will remove. 👍